### ISSUE TITLE
Reduce image size

### ISSUE THREAD

**Issue by MikhailSadovsky** [ostis-apps/dockerized-ostis#12]
Check options to reduce total image size. It's probably needed to review all dependencies and remove redundant

**Comment by illfate** [comment]
Better to use lightweight linux base like alpine

**Comment by illfate** [comment]
For compiling can be used docker multistage build

**Comment by PlagaMedicum** [comment]
Yes, we thought about this, but we're not sure that it will really help and don't break something + truth be told, some scripts in main repositories need to be refactored.

**Comment by FallenChromium** [comment]
Alright, so I made a little experiment and tried to rewrite the Dockerfile into multi-stage Alpine Linux based build. I want to inform you that I've encountered several interesting problems with Alpine:
1. Many dependencies are not in the official package base: antlr, rocksdb, future, etc.
2. Installing Python to this container made it waay bigger (around 72MB compared to 5MB stock)
3. ...and installing Python deps took waay more time and space than I've initially expected. That's because Alpine uses musl, not libc, so you can't just download PyPI's wheel for installing pip packages, you have to actually compile all of them (except those in the package manager, but you can't manage versions here, and either way there aren't many python packages). Yikes. 

In the end, building sc-machine + sc-web with the current dependencies set is possible, but not feasible on Alpine Linux. 

If we're trying to balance between size, build time and convenience, I'd lean to something like [minideb](https://github.com/bitnami/minideb) (minideb + python is just 79MB! And we still have access to apt packages on this distro)...
...OR, we could use pure Debian, which also isn't too far off. I think using Debian for this rewrite will be our safe bet, and in case we would like to swap the base to something else, the script will remain largely unchanged due to the package manager and package list being identical.

P.S While researching I've found a great tool named [exodus](https://github.com/intoli/exodus), made for creating portable binaries that carry dynamically linked libraries with them. If anyone would decide to move to Alpine in the future (and probably when Python version of sc-web will be replaced at some moment in the future), we could use it to port sc-machine executable to almost any Linux distribution. It's just a side note though.
